### ISSUE TITLE
iscsi module does not include iscsi-init.service

### ISSUE THREAD

**Issue by gonzoleeman** [dracutdevs/dracut#1537]
**Describe the bug**
The 95iscsi/module-setup.sh script lists most of the open-iscsi systemd service files, but does not include the newer iscsi-init.service unit.

**Distribution used**
Seen in SLES/openSUSE, but undoubtedly present in all distros.

**Dracut version**
049.1+suse.187.g63c1504f (but also present in the current latest code, based upon inspection)

**Init system**
systemd

**To Reproduce**
Boot with iscsi root and rd.debug enabled, and you will see error messages about iscsi-init.service not present so not called.

**Expected behavior**
The iscsi-init service is called if and only if there is no /etc/iscsi/initiatorname.iscsi. So we should see no messages.

**Additional context**
This script must be installed by dracut (iscsi-init.service) or we will see error messages. Other than the error messages, I don't believe this bug has any other negative side effect. But it is simple to fix.

**Comment by gonzoleeman** [comment]
The fix for this seems simple:

```
diff --git a/modules.d/95iscsi/module-setup.sh b/modules.d/95iscsi/module-setup.sh
index 88d6c92baea4..4b21fbb90118 100755
--- a/modules.d/95iscsi/module-setup.sh
+++ b/modules.d/95iscsi/module-setup.sh
@@ -192,6 +192,7 @@ install() {
     inst_multiple -o \
         "$systemdsystemunitdir"/iscsid.socket \
         "$systemdsystemunitdir"/iscsid.service \
+        "$systemdsystemunitdir"/iscsi-init.service \
         "$systemdsystemunitdir"/iscsiuio.service \
         "$systemdsystemunitdir"/iscsiuio.socket \
         "$systemdsystemunitdir"/sockets.target.wants/iscsid.socket \

```

I will submit a pull request, unless advised otherwise.

**Comment by johannbg** [comment]
Well that upstream type unit file wont work on a stateless system anyways make the PR so we can include it or make note that you wont and I will.

**Comment by gonzoleeman** [comment]
I have updated my branch to pull in the iscsi-init.service file.

**Comment by gonzoleeman** [comment]
It looks like this was fixed with commit bb6770f1a413b ("fix(iscsi): add iscsid.service requirements"), so I'll close this issue. Thank you!
